.circleci/parse_yaml.py

```diff
@@ -0,0 +1,269 @@
+
+"""Reading and parsing through the contributor's yaml file.
+(c) YoSon Park
+
+This is the default uploading skim for single and bulk files 
+using zenodo. For PR-based file uploader, check .circleci/deploy.sh 
+and .circleci/process_submitted_data.py
+"""
+
+
+import sys
+import yaml
+import pandas
+import pathlib
+import logging
+
+
+CONFIG_FILE_EXTENSION = ('yml', 'yaml')
+# transparent compressions recognized by tarfile 'r:*' are:
+# gzip, bz2, and lzma (xz)
+COMPRESSION_EXTENSION = ('gz', 'bz2', 'xz')
+BULK_EXTENSION = ('tgz', 'tar.gz', 'tar.bz2', 'tar.xz')
+FILE_EXTENSION = ('txt', 'tsv')
+
+logging.basicConfig(level='INFO')
+
+
+def check_yaml_before_commit():
```

Comment:
This function and the one follows (`check_yaml_in_ci`) can be removed, unless you plan to use it later.

Reply:
Yes, I plan to use them later and also locally